packages/plugins/src/foundry/utils/index.ts

```diff
@@ -74,12 +77,43 @@ export const getContractArtifact = async (
  */
 export const makeGetConfigArtifacts = (
   artifactFolder: string,
-  buildInfoFolder: string
+  buildInfoFolder: string,
+  cachePath: string
 ): GetConfigArtifacts => {
   return async (contractConfigs: UserContractConfigs) => {
+    // Check if the cache directory exists, and create it if not
+    if (!(await existsAsync(cachePath))) {
+      fs.mkdirSync(cachePath)
+    }
+
+    const buildInfoCacheFilePath = join(cachePath, 'chugsplash-cache.json')
+    let buildInfoCache: Record<
+      string,
+      {
+        name: string
+        time: number
+        contracts: string[]
+      }
+    > = fs.existsSync(buildInfoCacheFilePath)
+      ? JSON.parse(await readFileAsync(buildInfoCacheFilePath, 'utf8'))
+      : {}
+
     const buildInfoPath = join(buildInfoFolder)
 
-    const buildInfoPromises = fs
+    // If the required files are missing then clear the cache
+    // We only check the first file because we can safely assume that if any file is missing then
+    // the first one will also be missing due to how the build info files are generated.
+    // The only case where this wouldn't be true is if the user manually deleted the file, which
+    // is not a legitimate use case.
+    if (
```

Comment:
Is there a place in this function where it'd be convenient to make this check on every build info in the cache, and not just the first one? This seems fine, but it'd be great if we could check this in a less hacky way

Reply:
We pretty much already do that. When we read the files, we detect if one is missing and throw an error recommending the user recompile with force. We could also do the check separately, but would be a wasted operation.